### PR TITLE
test(providers): add unit tests for resource name/ID extraction

### LINKED ISSUE #194: Add provider-level unit tests for resource name/ID extraction
## Summary

43 of 44 providers have zero test files. While full `InitResources()` testing requires SDK mocking (low ROI), many providers contain ARN/ID parsing and resource name extraction logic that is pure functions and easy to unit test.

## Pattern to follow

`providers/aws/msk_test.go` demonstrates the approach: extract ARN/ID parsing into named helpers, then test them with table-driven cases including edge cases.

## Suggested starting providers

- **AWS** — largest provider (92 source files), many resources with ARN parsing
- **GCP** — second largest (78 source files), project/resource ID extraction
- **Azure** — 38 source files, subscription/resource-group path parsing
- **IBM** — 54 source files, CRN parsing

## Scope

- Identify pure-function parsing logic in each provider
- Extract into testable helpers where inline
- Add table-driven tests with edge cases
- No SDK mocking or API call testing

### PR DESCRIPTION
**Pull request by chenrui333**
## Summary

Add unit tests across AWS, GCP, and Azure providers covering resource name and ID extraction logic.

### Source changes

- Extract `arnLastSegment` shared helper in `providers/aws/aws_service.go`
- Refactor `ecs.go`, `sns.go`, `sqs.go` to use `arnLastSegment` instead of inline `strings.Split` calls

### New test files

- `providers/aws/ecs_test.go` — `arnLastSegment` (shared helper covering ECS, SNS, SQS, and general ARN/URL parsing)
- `providers/aws/route53_test.go` — `wildcardUnescape`, `cleanZoneID`, `cleanPrefix` (production helpers)
- `providers/gcp/gcp_name_helpers_test.go` — GCP resource name conventions (last-segment, BigQuery dataset ID, KMS key ring ID composition, zone link)
- `providers/azure/helper_test.go` — `ParseAzureResourceID` (standard, lowercase, RG-only, error cases), `asHereDoc`

### Approach

AWS tests call extracted production helpers directly (`arnLastSegment`, Route53 helpers). Azure tests call the existing `ParseAzureResourceID` production code. GCP tests validate the naming convention used across generated files, documented as convention guards.

Closes #194

### CODE REVIEW

File: providers/gcp/gcp_name_helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		fullID   string
+		wantName string
+	}{
+		{"standard dataset", "project:my_dataset", "my_dataset"},
+		{"dataset with numbers", "project:dataset_2024", "dataset_2024"},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			got := strings.Split(tc.fullID, ":")[1]
```
[the remark this answers is not in the document]
Comment (chenrui333) [reply providers/gcp/gcp_name_helpers_test.go]: @codex review again
